AutoSteamApp/Automaton/SteamworkAutomaton.cs

```diff
@@ -0,0 +1,348 @@
+        public void Run(CancellationToken cts)
+        {
+            try
+            {
+                while (!cts.IsCancellationRequested)
+                {
+                    if (_SupportedVersion && !ConfigurationReader.RandomRun)
+                    {
+                        // If we have satisfied all exit conditions
+                        if (CheckForExitCondition())
+                            // If we should auto exit, quit immediately
+                            if (ConfigurationReader.ShouldAutoQuit)
+                                Environment.Exit(0);
+                            // Otherwise wait for quit to be typed
+                            else
+                                return;
+
+                        // Otherwise we need to extract the sequence
+                        ExtractAndEnterSequence(cts);
+
+                        // Then check the steamworks state
+                        CheckSteamworksState(cts);
+                    }
+                    else
+                    {
+                        EnterRandomSequence(cts);
+                    }
+                }
+            }
+            catch (Exception e)
+            {
+                Log.Exception(new Exception("Failed automating steamworks\n\t", e));
+                Log.Warning("Something went wrong trying to automate the steamworks.");
+                // If we should auto exit, quit immediately
+                if (ConfigurationReader.ShouldAutoQuit)
+                    Environment.Exit(-1);
+                // Otherwise wait for quit to be typed
+                else
+                    return;
+            }
+        }
+
+        private void EnterRandomSequence(CancellationToken cts)
+        {
+            // Generate a sequence to input
+            VirtualKeyCode[] sequence;
+            // If the version is unsuported, use a random sequence
+            sequence = StaticHelpers.RandomSequence();
+            // Press the buttons and wait 30ms then press start (in case the cutscene plays)
+            // Press the keys
+            Log.Debug("Random Sequence: [" + string.Join(", ", sequence.Select(x => x.ToString())) + "]");
+            // Wait until we have focus
+            if (!_Process.HasFocus())
+                Log.Message("Waiting for MHW to have focus.");
+            while (!_Process.HasFocus() && !cts.IsCancellationRequested) { };
+
+            for (int i = 0; i < sequence.Length; i++)
+            {
+                StaticHelpers.PressKey(_InputSimulator, sequence[i], ConfigurationReader.RandomInputDelay);
+                Thread.Sleep(ConfigurationReader.RandomInputDelay);
+            }
+            StaticHelpers.PressKey(_InputSimulator, VirtualKeyCode.SPACE, ConfigurationReader.RandomInputDelay);
+            Thread.Sleep(ConfigurationReader.RandomInputDelay);
+            StaticHelpers.PressKey(_InputSimulator, VirtualKeyCode.VK_X, ConfigurationReader.RandomInputDelay);
```

Comment:
Why `X` ? if that the animation cancel key ? - if so, it should be configured as it is dependent to users configuration.